app/views/questionnaire.py

```diff
@@ -133,27 +135,29 @@ def post_block(routing_path, schema, metadata, collection_metadata, answer_store
         next_location = router.get_next_location()
         return _redirect_to_location(next_location)
 
-    block = _get_block_json(current_location, schema, answer_store, metadata)
+    block = schema.get_block(current_location.block_id)
 
     placeholder_renderer = PlaceholderRenderer(block, answer_store=answer_store, metadata=metadata)
     replaced_block = placeholder_renderer.render()
 
+    new_block = transform_variants(replaced_block, schema, metadata, answer_store)
```

Comment:
In `get_block` we call this `transformed_block` and here we use `new_block`
Should we be more consistent with this naming?